extensions/markdown-language-features/src/features/workspaceSymbolProvider.ts

```diff
@@ -26,10 +26,38 @@ class VSCodeWorkspaceMarkdownDocumentProvider extends Disposable implements Work
 
 	private _watcher: vscode.FileSystemWatcher | undefined;
 
-	async getAllMarkdownDocuments() {
-		const resources = await vscode.workspace.findFiles('**/*.md', '**/node_modules/**');
-		const docs = await Promise.all(resources.map(doc => this.getMarkdownDocument(doc)));
-		return docs.filter(doc => !!doc) as SkinnyTextDocument[];
+	/**
+	 * Reads and parses all .md documents in the workspace.
+	 * Files are processed in batches, to keep the number of open files small.
+	 *
+	 * @returns Array of processed .md files.
+	 */
+	async getAllMarkdownDocuments(): Promise<SkinnyTextDocument[]> {
+		const maxConcurrent = 20;
+		const ignoreList: (string | vscode.RelativePattern)[] = ['**/node_modules/**'];
+		const docList: SkinnyTextDocument[] = [];
+
+		let isDone = false;
+		while (!isDone) {
+			await new Promise(async (resolve) => {
+				const resources = await vscode.workspace.findFiles('**/*.md', `{${ignoreList.join(',')}}`, maxConcurrent);
+				if (resources.length < maxConcurrent) {
+					isDone = true;
+				}
+
+				await Promise.all(resources.map(doc => {
+					// Convert path to be relative from the workspace
```

Comment:
Why is this  converationneeded?

Reply:
`workspace.findFiles` is relative to the workspace, so I couldn't just turn around and pass back in `doc.fsPath` as an exclude.
`/Users/nrayburn/documents/vscode/contributing.md` vs `contributing.md` for example.  Maybe there is a better way to do this that I missed, but this was the only way I could find to get it working.

This becomes irrelevant, if we only do the `workspace.findFiles` call once though.